### ISSUE TITLE
Sprint Poker v1 vs. Target Scoping

### ISSUE THREAD

**Issue by jordanh** [ParabolInc/parabol#3993]
### Acceptance criteria
- v1 vs target AC drafted
- reviewed together as a team
- break down into next-step issues for upcoming sprints
### Estimated effort
- 3 hours
### For quick reference
- [The latest concepts on parade in Figma](https://www.figma.com/proto/csERjrh67KZKgnT5VahFSb/Sprint-Poker?node-id=424%3A738&viewport=428%2C142%2C0.25&scaling=scale-down), used in the user research 📖 
- [Grouped concerns and highlights based](https://action.parabol.co/meet/eGghuEJpte/discuss/1) on the user research 🔒 
- The [user research docs](https://www.notion.so/parabol/Planning-Meeting-User-Research-3f70fc63a86647d4bd0492a6f0303e9b) in Notion 🔒 
- General [UX design notes](https://www.notion.so/parabol/Sprint-Poker-2bf25b59fd044b82b8318fc39bbbe60f) in Notion 🔒 
---
# Acceptance criteria draft WIP
## TL;DR for V1
- **Re: general concerns**, we need to explore a secondary CTA, perhaps in the lobby, that nudges a user to prepare for the meeting in advance. In effect, it probably starts the meeting but do we silence the alerts in the meeting menu and snackbar? Are some phases of a pre-meeting type?
- **Re: templates**, we provide a single dimension but multiple scale options
- **Re: adding and importing stories**, we decide on A or B:
  A) The ability to add stories as Parabol tasks that can be pushed to Jira or GitHub and become read-only
  B) The ability to both add stories that can be pushed and the ability to import read-only from Jira and/or GitHub with some basic filtering like text string query or label
- **Re: voting and reviewing**, folks can vote on and review a single dimension. Folks will mostly like just discuss and enter a final vote before they revote. Folks can change their vote at any time.
- **Re: summary and exporting**, we provide some meta (velocity, total points, participants, etc.) and a table that has a row for each story that includes title and value. The table can be exported as a CSV. We decide how soon we push out story point values as comments on issues or to a unique field.
### Team velocity and total story points
Both of these could make it into V1:
- The team can update the current velocity in the left navigation. TBD is this meta stored on the team or the meeting?
- The running total is shown somewhere in the left navigation where the stories are sorted.
### Scoping which issues to estimate as meeting pre-work
This will take some design work that could make it into V1 or be deferred:
- Users should probably scope the issues to estimate ahead of time. One potential UX:
  - A lead starts the meeting alone
    - Perhaps a secondary CTA is ‘Set up the meeting’ or ‘Prepare the meeting’ or ‘Add stories for the meeting’
  - The lead creates or imports issues to estimate alone or maybe with 1 other role
  - The lead invites the team to the active meeting later
  - User job story: _When I’m planning to use Parabol so that my team can estimate the points of stories for a sprint in a quick and easy fashion, I want a way to set up for the meeting ahead of time so that we have a list of stories ready to go when we start the meeting_.
### General meeting IA & UI
No change:
- The bottom bar has Tips, Timer, Next, and End Meeting. The components for adding, importing, voting, etc. are in the main view.
- The meeting can be renamed
- Folks can be invited to the meeting. Participants see active viewers in the top right
- We anticipate adding video one day using the general meeting IA patterns we’re working toward. This may have implications as to how we lay out the thread component
### TL;DR
Most folks only use a single dimension. There’s no need to create many templates just yet. Let’s expose a single template where folks can edit the name of the dimension and select or edit scales. We budget our first efforts to build out scales. This will also make the V1 of the Estimate Phase easier since folks only vote on and review a single dimension.
### V1
- In the UI…
  - There’s a single template with a single dimension ‘Story Points’ with a default scale selected
  - Users can edit the name of the dimension
  - Users can select or clone scales we provide
  - Users can create new scales 
  - Users can select, edit, and delete scales they cloned or created
- In the backend…
  - There can be multiple templates, both team-owned and public, Parabol-owned
  - There can be multiple scales, both team-owned and public, Parabol-owned
  - Templates can have multiple dimensions, each dimension can have a selected scale
### Target
- In the UI…
  - Users can have multiple templates on the team
  - Templates can have 1 or many dimensions
  - Users can use or clone templates shared by the org
  - Users can clone public templates. **TBD** what about simply using templates provide by Parabol?
  - Users can share templates with their organization or with the public
  - Users can optionally include Parabol-owned formulas to templates for weighing and ranking stories. E.g. users can define 1 or many value dimensions divided by 1 or many cost dimensions to get a weighted shortest job first value. Value and cost are probably set as types of dimensions. When selecting this option, all dimensions require using the same scale with a set of integer values like 1-5, or 1-3-5, or Fibonacci in order to do fair maths.

### TL;DR
We start with Parabol tasks that can be pushed to Jira or GitHub one way. We build toward having integrations that allow for granular, API-based filtering and 2-way synchronization.
### V1
- In the UI…
  - Users can add new tasks as stories to be estimated in the main view
  - Users can edit tasks if they’re not pushed to Jira or GitHub
  - Users can push those tasks to Jira or GitHub
  - Users can sort the order of tasks in the main view
  - Users can sort the order of tasks in the left navigation, like agenda items
  - Users can add new tasks inline in the left navigation, just writing in a bit of text, like agenda items
- In the backend…
  - Tasks created are added as stage items for the Estimate Phase
  - Users can remove stories as tasks in the main view or in the left navigation, like agenda items
### Next
- In the UI…
  - Users can add a Jira integration and see read-only issues as tasks in Parabol
  - Users can use the most common filters from Jira e.g. label or text string query
  - Users see a list view of the integrated issues. Tapping on an issue expands to show the top comment/description
  - Users can tap on a link per integrated task to see the issue in Jira in a new tab
  - Users can use multi-select or a ‘select all’ accordance to add queried Jira issues to the Estimate Phase
  - We only write to the issue at the end where we add the story points as a comment or to a unique field
### Target
- In the UI…
  - Users can add a Jira or GitHub integration (maybe more?) and read and edit 2-way synced issues
  - Users can query by more granular, API-based filtering as we hear from users
  - Users can tap to expand an integrated issue and maybe see more than just the first comment as a description; can they see comments? Attachments? Other fields? As we go we can discover if there’s real value here among our users besides what we heard from our early user research.

### TL;DR
Since most folks (we’ve heard from) estimate a single dimension, we can design a progressive view that takes a team from voting and reviewing a single story along a single dimension to voting and reviewing multiple dimensions. We can build out the simplest version first.
### V1
- A story meeting stage can be in voting or reviewing mode
- Users can vote on a single dimension
- Users can move ahead and vote for all stories using the async before sync principle
- Users can update their vote at any time during voting or reviewing mode
- The votes are automatically revealed when all participants have voted
- Users can review the estimation for a single dimension once it’s been revealed
- We show the mode (most used value in voting)
- We highlight values that aren’t the mode
- The facilitator can manually reveal at anytime
- The facilitator can call a revote (this is a secondary affordance as they’ll most likely just have a discussion and agree to a final value)
- The facilitator can play the pass card to abstain from voting
- The facilitator can enter a final value for the story. How do we make this feel required?
- The discussion thread is available to the participants per story (meeting stage)

### Target
- Users can vote across several dimensions, 1 at a time
- The participants can review 1 dimension at a time
- The participants can expand to see the votes for all dimensions
- The facilitator can call a revote for a single dimension or all dimensions
- TBD the final value is written to the integrated issue as a comment or unique field instantly or at the end of the meeting?
- TK animation as multiplayer feedback (this needs a drawing and maybe a quick [Principle](https://principleformac.com/
) concept)

### TL;DR
We provide a table of all the stories and their estimated values. Folks want to know their velocity and the total points for all stories estimated. These are probably safe for V1. What we define as the target depends on how we increase the fidelity of template and integration options.
### V1
- The summary provides
  - A value for the team’s velocity
  - A value for the total story points
  - Meta like the number of stories and the list of participants
  - A table with a row for each story including the title and the final value
  - The table can be exported as CSV
### Target
- The summary provides
  - Alternative views for totals per dimension or by a formula for weighing stories when those templates are used
- Each integration has a designated place where the estimation is added, e.g:
  - In Jira a single dimension might be mapped to a unique story points field on the issue. This mapping would be part of the integration settings
  - In GitHub, we could add a comment with a small table or value and a link back to the story stage and/or meeting summary
- Gamification and analytics
  - How to potentially show streaking or stats
  - In the far fuzzy future, what insights can we provide about how a team estimates? It seems a lot of folks look for these trends in Jira so it’s probably not an interesting space for us.

**Comment by ackernaut** [comment]
I’ve moved [a copy of this draft to Notion](https://www.notion.so/parabol/Sprint-Poker-V1-vs-Target-Scoping-80b2ed9dfd8e4048bf183a19ee5f4326) so we can review async before our sync meeting

**Comment by ackernaut** [comment]
Closing this issue. We’ve agreed on some points regarding V1 versus V-next and have a list of issues to write to move us forward.
